### PR TITLE
Notifications: mark as read/unread and pin

### PR DESCRIPTION
**Pull request by andreynering**

### CODE REVIEW

**Comment by lunny** [comment]
What's the pinned? And How to remove pinned?

**Comment by andreynering** [comment]
I'm not good at UI, so suggestions and help are very welcome.

@lunny A pinned notification will always show. It will not be automatically marked as read when you open the issue, only when you click the "Mark as read" button.

**Comment by lunny** [comment]
LGTM

**Comment by Bwko** [comment]
LGTM

File: routers/user/notification.go [hunk 1]
```diff
@@ -79,3 +83,32 @@ func Notifications(c *context.Context) {
...
+
+	switch statusStr {
+	case "read":
+		status = models.NotificationStatusRead
+	case "unread":
+		status = models.NotificationStatusUnread
+	case "pinned":
+		status = models.NotificationStatusPinned
+	default:
+		c.Handle(500, "InvalidNotificationStatus", errors.New("Invalid notification status"))
```
Comment (ethantkoenig) [inline routers/user/notification.go:103]: Minor, but we should include the invalid status (i.e. `statusStr`) in the error message

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

**Review by appleboy — APPROVED** [review]

**Comment by gsantner** [comment]
@andreynering 
Note: "Use relative URLs" did not fix the URL in the top navigation bar.
Points to `mydomain.xyz/notifications` instead of  `mydomain.xyz/gitea/notifications`  (with setted up /gitea/ as subdir). Got the latest master, and everything else points correctly to /gitea/

**Comment by andreynering** [comment]
@gsantner I couldn't find the link you cited. The topbar link should be OK:

https://github.com/go-gitea/gitea/pull/629/commits/993c7d65f559bd95af70e9c06175ef4470990204#diff-675f585d9cd6a202ba88a50dc67e52ffR85

**Comment by gsantner** [comment]
@andreynering  Exactly at the line you pointed out, there is no "/gogs/" or "/gitea/" prefix. Yeah it should according to code, but it isn't.

grafik
(As you can see, everything else links correctly to /gogs/\*)

I think it should be `{{AppSubUrl}}`, not `{{$.AppSubUrl}}`. If you look some lines above in template code, you can see theres no `$.`

____

Should fix it:
`
<a href="{{AppSubUrl}}/notifications" class="ui head link jump item poping up" data-content='{{.i18n.Tr "notifications"}}' data-variation="tiny inverted">
`

`<a href="{{AppSubUrl}}/notifications?q=unread">
`

`<a href="{{AppSubUrl}}/notifications?q=read">
`

**Comment by andreynering** [comment]
@gsantner Oh, you're right. Thanks for pointing out. It will be fixed as part of https://github.com/go-gitea/gitea/pull/660

**Pull request merged**